### PR TITLE
Add ResNet training code

### PR DESCRIPTION
**Pull request by yuyu2172**
~~Merge after #652~~
~~Merge after #432, #435, #321~~

~~Aims at reproducing scheme used in https://github.com/facebook/fb.resnet.torch.~~
Edit:
The training code reproduces the scheme introduced below.
This performs better than `fb.resnet.torch` and scales to the situation with many GPUs.
https://arxiv.org/pdf/1706.02677.pdf

- [x] *Linear Scaling Rule* for lr (see the paper)
- [x] Set default value of gamma for BNs in the end of bottleneck to 0
- [x] use corrected momentum SGD
- [x] Have correct scale for weight decay (ChainerMN accomplishes this by default)
- [x] Learning rate scheduling happens at 30, 60, 80-th epoch.
- [x] stop using color related data augmentation

Edit 11/16:
Although "ResNet in 1 hour" paper is new, the training scheme is adapted by some researchers (e.g. [mixup](https://arxiv.org/abs/1710.09412)).

### CODE REVIEW

**Comment by yuyu2172** [comment]
The result without setting `gamma` to 0 for the last BN of each ResBlock.

```
8 GPU
23.60

32 GPU
24.37
```

**Comment by yuyu2172** [comment]
I updated the code with lr warmup. The behavior stays the same when 8GPUs are used.

results:
ResNet50 with 32GPUs
23.54

**Comment by yuyu2172** [comment]
@Hakuyume 
Please review this

**Comment by Hakuyume** [comment]
Both `a weight` and `weights` are used for "a set of parameters of a model". Which one is better?

**Comment by yuyu2172** [comment]
I am checking how slow the training gets without options like `autotune`.

**Comment by yuyu2172** [comment]
Thanks for reviewing.
I reflected ours discussions to the code.

File: examples/classification/README.md [hunk 1]
```diff
@@ -32,15 +34,49 @@ Ten crop error rate.
...
+
+The scores of the models trained with `train_imagenet_multi.py`, which can be executed like below.
+Please consult the full list of arguments for the training script with `python train_imagenet_multi.py -h`.
+```
+$ mpiexec -n N python train_imagenet_multi.py <path_to_train_dataset> <path_to_val_dataset>
+```
+
+The training procedure carefully follows the "ResNet in 1 hour" paper [5].
+
+##### Performance tip
```
Comment (Hakuyume) [inline examples/classification/README.md]: Why did you use 5th level? (4th level `####` is not enough?)

File: examples/classification/train_imagenet_multi.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+
+class TrainTransform(object):
+
+    def __init__(self, mean):
+        self.mean = mean
+
+    def __call__(self, in_data):
+        img, label = in_data
+        _, H, W = img.shape
```
Comment (Hakuyume) [inline examples/classification/train_imagenet_multi.py]: `H` and `W` are not used?

File: examples/classification/README.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Classification
 
-## Performance
+## ImageNet
+
+### Weight conversion
 
 Single crop error rate.
```
Comment (Hakuyume) [inline examples/classification/README.md]: For the consistency with `Trained model`, it would be nice to add something like `~ of the models converted from Caffe model`.

File: examples/classification/README.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # Classification
 
-## Performance
+## ImageNet
```
Comment (Hakuyume) [inline examples/classification/README.md]: `he` and `fb` should be distinguished from each other in score board.
